Synchronizing differences between internal repository and open source

BA-MBSE opened the pull request:
+ Adding authentication delegation for teamwork cloud
+ Adding changes for MDK compatibility

HuiJun commented on mms-ent/repo-amp/src/main/java/gov/nasa/jpl/view_repo/db/PostgresHelper.java:
```diff
@@ -1451,6 +1451,26 @@ public void deleteEdges(String parentSysmlId, DbEdgeTypes edgeType) {
+                nullParents.add(rs.getString(1));
+            }
+        } catch (NullPointerException npe) {
+        } catch (Exception e) {
```
Might want to put a comment in to let people know this catch should be empty

BA-MBSE committed:
Synchronizing differences between internal repository and open source
+ Adding authentication delegation for teamwork cloud
+ Adding changes for MDK compatibility

BA-MBSE committed:
Fixed exception imports

HuiJun commented on core/core.gradle:
```diff
@@ -13,6 +13,7 @@ dependencies {
     api commonDependencies.'log4j'
     api commonDependencies.'log4j-api'
     api commonDependencies.'swagger-annotations'
+    api 'org.json:json:20190722'
```
This dependency and JsonUtils seems unnecessary, in core at least.

BA-MBSE replied:
Yeah, we can move it

HuiJun commented on core/src/main/java/org/openmbee/sdvc/core/utils/RestUtils.java:
```diff
@@ -33,4 +37,15 @@ public String getCookieValue(ResponseEntity<?> responseEntity, String cookieName
     }
+    
+    
+    public HttpHeaders basicAuthHeader(String username, String password){
```
This may go against our stateless architecture design goals. Why do we need to set cookies? If necessary, can we move this where it's utilized, instead of in core?

BA-MBSE replied:
This is just setting a basic auth header.  No cookies are being set.
The cookie stuff in the previous method is pulling cookies out of a Response  - this is used for getting authentication details from Teamwork Cloud.  We can move it, but it seems generic enough for core.

HuiJun replied:
Sorry, I was referring to the file as a whole. I'm thinking we could move RestUtils to authenticator instead of having them in core.

dlamoris replied:
I don't think twc imports the auth module - seems like they're only used by twc anyways, may be better to just move to twc

BA-MBSE replied:
Ok, moved to twc module

BA-MBSE committed:
Moved JSON utils to twc module, various cleanup

BA-MBSE committed:
Moved RestUtils to twc module
